src/katsdpcontroller/generator.py

```diff
@@ -1513,10 +1489,140 @@ def get_baseline_index(a1, a2):
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-chans",
+                "The number of frequency channels in this stream's overall output",
+                default=stream.n_chans,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-chans-per-substream",
+                "Number of channels in each substream for this data stream",
+                default=stream.n_chans_per_substream,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-vengs",
+                "The number of V-engines in the instrument",
+                # TODO: Clarify how this is deciphered. One v-eng to one stream?
+                default=stream.n_vengs,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.veng-out-bits-per-sample",
+                "V-engine output bits per sample. Per number, not complex pair- "
+                "Real and imaginary parts are both this wide",
+                default=stream.bits_per_sample,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.power-int-time",
+                "Interval over which power is measured and normalisation is performed",
+                "s",
+                # default=stream.power_int_time,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.fir-taps",
+                "Number of taps in the rational downconversion filter",
+                # default=?
+                initial_status=Sensor.Status.Nominal,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.sideband-taps",
+                "Number of taps in the filter used to split into side-bands",
+                # default=?
+                initial_status=Sensor.Status.Nominal,
+            ),
+            Sensor(
+                str,
+                f"{stream.name}.pol-ordering",
+                "Which polarisations are produced and the order of their thread IDs. "
+                "Each element is one of 'x', 'y', 'R' or 'L'.",
+                # default="" ?
+                initial_status=Sensor.Status.Nominal,
+            ),
+        ]
+
+        for ss in stream_sensors:
+            g.graph["stream_sensors"].add(ss)
+
+        # TODO: init_telstate
+
+    for i in range(n_engines):
+        vgpu = ProductLogicalTask(f"v.{base_name}.{i}", streams=streams, index=i)
+        vgpu.subsystem = "cbf"
+        vgpu.image = "katgpucbf"
+        vgpu.fake_katcp_server_cls = None  # TODO: No fake server yet
+
+        vgpu.sensor_renames[f"{stream.name}.delay"] = f"{stream.name}.{i}.delay"
```

Comment:
The engine won't include the stream name in sensors, and we don't need to index the engines because there is only one. I always get a bit confused about which way around sensor_renames works, but IIRC, this should then be

```suggestion
        vgpu.sensor_renames["delay"] = f"{stream.name}.delay"
```